packages/dd-trace/test/plugins/agent.js

```diff
@@ -381,37 +389,85 @@ module.exports = {
     }
   },
 
-  // Register handler to be executed each agent call, multiple times
+  /**
+   * Register handler to be executed each agent call, multiple times
+   * @param {Function} handler
+   */
   subscribe (handler) {
-    traceHandlers.add({ handler })
+    traceHandlers.add({ handler }) // TODO: SHOULD BE .add(handler) SO WE CAN DELETE
   },
 
-  // Remove a handler
+  /**
+   * Remove a handler (TODO: THIS DOES NOTHING)
+   * @param {Function} handler
+   */
   unsubscribe (handler) {
     traceHandlers.delete(handler)
   },
 
   /**
-   * Register a callback with expectations to be run on every tracing payload sent to the agent.
+   * Callback for running test assertions against traces.
+   *
+   * @callback testAssertionTracesCallback
+   * @param {Array.<Array.<span>>} traces - For a given payload, an array of traces, each trace is an array of spans.
+   */
+
+  /**
+   * Callback for running test assertions against a span.
+   *
+   * @callback testAssertionSpanCallback
+   * @param {span} span - For a given payload, the first span of the first trace.
+   */
+
+  /**
+   * This callback gets executed once for every payload received by the agent.
+   * It calls the callback with a `traces` argument which is an array of traces.
+   * Each of these traces is an array of spans.
+   *
+   * @param {testAssertionTracesCallback} callback - runs once per agent payload
+   * @param {Object} options
+   * @returns Promise
    */
   assertSomeTraces (callback, options) {
-    return runCallback(callback, options, traceHandlers)
+    return runCallbackAgainstTraces(callback, options, traceHandlers)
+  },
+
+  /**
+   * Same as assertSomeTraces() but only provides the first span (traces[0][0])
+   * This callback gets executed once for every payload received by the agent.
+
+   * @param {testAssertionSpanCallback} callback - runs once per agent payload
+   * @param {Object} options
+   * @returns Promise
+   */
+  assertFirstTraceSpan (callback, options) {
+    return runCallbackAgainstTraces(function (traces) {
+      return callback(traces[0][0])
+    }, options, traceHandlers)
   },
 
   /**
    * Register a callback with expectations to be run on every stats payload sent to the agent.
    */
   expectPipelineStats (callback, options) {
-    return runCallback(callback, options, statsHandlers)
+    return runCallbackAgainstTraces(callback, options, statsHandlers)
   },
 
-  // Unregister any outstanding expectation callbacks.
+  /**
+   * Unregister any outstanding expectation callbacks.
+   */
   reset () {
     traceHandlers.clear()
     statsHandlers.clear()
   },
 
-  // Stop the mock agent, reset all expectations and wipe the require cache.
+  /**
+   * Stop the mock agent, reset all expectations and wipe the require cache.
+   * @param {Object} opts
+   * @param {boolean} opts.ritmReset - If true, resets the Require In The MIddle cache, useful for TODO.
```

Comment:
@rochdev can you explain situations where a test author would want to use these two flags? I'd like to document them here so we get the reasons when autocompleting.

Reply:
Same for `wipe`. They're both things we shouldn't be doing.